Lib/subprocess.py

```diff
@@ -978,6 +979,19 @@ def __init__(self, args, bufsize=-1, executable=None,
 
             raise
 
+    def __repr__(self):
+        max_args_length = 30
+        returncode = self.returncode or 'unfinished'
+        args = ' '.join(map(shlex.quote, self.args))
+
+        if len(args) > max_args_length:
+            args = f"{args[:max_args_length]}...".rstrip()
```

Comment:
`.rstrip()` will never do anything here.

Also, this should use `[:max_args_length - 3]` to compensate for the added '...'.
```suggestion
            args = f"{args[:max_args_length - 3]}..."
```